src/runtime_src/core/edge/user/aie/aie.cpp

```diff
@@ -113,25 +115,30 @@ open_context(const xrt_core::device* device, xrt::aie::access_mode am)
 {
 #ifndef __AIESIM__
   auto drv = ZYNQ::shim::handleCheck(device->get_device_handle());
+  std::lock_guard access_lock_guard(am_lock);
+  if (access_mode != xrt::aie::access_mode::none) {
```

Comment:
Does this test require to hold the test longer?

Reply:
Actually what this uncommented code does?

Reply:
I realize that neither the `.h` nor the `.cpp` have comments on their functions. This is not very inclusive for an open-source project.

Reply:
Added comments